runtime/onert/core/src/ir/train/TrainableGraph.cc

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ir/train/TrainableGraph.h"
+
+namespace onert
+{
+namespace ir
+{
+namespace train
+{
+
+TrainableGraph::TrainableGraph(const Graph &graph) : _graph{graph}, _operations{} {}
+
+TrainableGraph::TrainableGraph(Graph &graph, const TrainableOperations &toperations)
+  : _graph{graph}, _operations{}
+{
+  graph.operations().iterate([&](const onert::ir::OperationIndex &idx, onert::ir::Operation &op) {
+    const auto &trainable_op = toperations.at(idx);
+    appendTrainableOperationOnly(idx, trainable_op.clone(op));
+  });
+}
+
+TrainableGraph::~TrainableGraph(void) = default;
```

Comment:
(Out of curiosity) Why defaulted dtor is defined in `cc`, not in `h`. I usually put this in header.

Reply:
I don't know about that. 
Graph.h is also like that, so I just copied it.

Reply:
This was moved in the header file.